bundles/event_admin/event_admin/src/celix_event_admin.c

```diff
@@ -415,6 +454,131 @@ int celix_eventAdmin_removeEventHandlerWithProperties(void* handle, void* svc, c
     return CELIX_SUCCESS;
 }
 
+int celix_eventAdmin_addRemoteProviderService(void* handle, void* svc, const celix_properties_t* props) {
+    assert(handle != NULL);
+    assert(svc != NULL);
+    celix_event_admin_t* ea = handle;
+    long serviceId = celix_properties_getAsLong(props, CELIX_FRAMEWORK_SERVICE_ID, -1L);
+    if (serviceId < 0) {
+        celix_logHelper_error(ea->logHelper, "Remote provider service id is invalid.");
+        return CELIX_ILLEGAL_ARGUMENT;
+    }
+    celix_auto(celix_rwlock_wlock_guard_t) wLockGuard = celixRwlockWlockGuard_init(&ea->lock);
+    celix_longHashMap_put(ea->remoteProviderServices, serviceId, svc);
+    return CELIX_SUCCESS;
+}
+
+int celix_eventAdmin_removeRemoteProviderService(void* handle, void* svc, const celix_properties_t* props) {
+    assert(handle != NULL);
+    assert(svc != NULL);
+    celix_event_admin_t* ea = handle;
+    long serviceId = celix_properties_getAsLong(props, CELIX_FRAMEWORK_SERVICE_ID, -1L);
+    if (serviceId < 0) {
+        celix_logHelper_error(ea->logHelper, "Remote provider service id is invalid.");
+        return CELIX_ILLEGAL_ARGUMENT;
+    }
+    celix_auto(celix_rwlock_wlock_guard_t) wLockGuard = celixRwlockWlockGuard_init(&ea->lock);
+    celix_longHashMap_remove(ea->remoteProviderServices, serviceId);
+    return CELIX_SUCCESS;
+}
+
+static void celix_eventAdmin_retrieveLongTimeUnusedEventSeqIdCache(celix_event_admin_t* ea) {
+    if (celix_stringHashMap_size(ea->eventSeqIdCache) > 16) {
+        celix_string_hash_map_iterator_t iter = celix_stringHashMap_begin(ea->eventSeqIdCache);
+        while (!celix_stringHashMapIterator_isEnd(&iter)) {
+            celix_event_seq_id_cache_t* cache = iter.value.ptrValue;
+            if (celix_elapsedtime(CLOCK_MONOTONIC, cache->lastModified) > 60*60/*1h*/) {
```

Comment:
I think it would be better to make the 1h threshold configureable using a config property.